Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java

```diff
@@ -17,79 +20,150 @@
+                List<INetworkNodeInfo> localNetworkNodeInfoList =
+                        networkManager.getNetworkTopNodes(network.getNetworkId());
+                if (localNetworkNodeInfoList != null) {
+                    networkNodeInfoList.addAll(localNetworkNodeInfoList);
+                }
+            } catch (QuadrigaStorageException e) {
+                throw new QuadrigaStorageException("Database Error while getting network top nodes for a network with ID: " +
+                        network.getNetworkId(), e);
+            }
+        }
+
+        ITransformedNetwork networkJSon = transformer.transformNetwork(networkNodeInfoList);
+        return networkJSon;
+    }
+
+    @Override
+    public ITransformedNetwork searchTransformedNetworkOfProject(String projectId, String conceptId)
+            throws QuadrigaStorageException {
+        List<INetwork> networkList = getNetworkList(projectId);
+        if (networkList == null) {
+            return null;
+        }
+
+        Map<String, Node> searchNodes = new HashMap<String, Node>();
+        List<Link> searchLinks = new ArrayList<Link>();
+        for (INetwork network : networkList) {
+            try {
+                List<INetworkNodeInfo> localNetworkNodeInfoList =
+                        networkManager.getNetworkTopNodes(network.getNetworkId());
+                // check if this network has the node with given concept id
+                if (localNetworkNodeInfoList != null) {
+                    ITransformedNetwork transformedNetwork =
+                            transformer.transformNetwork(localNetworkNodeInfoList);
+                    // search for concept id
+                    Node searchNode = null;
+                    for (Node node: transformedNetwork.getNodes().values()) {
+                        if (node.getConceptId().equals(conceptId)) {
+                            searchNode = node;
+                            break;
+                        }
+                    }
+                    // if node does not exist continue
+                    if (searchNode == null) {
+                        continue;
+                    }
+
+                    Set<String> searchStatements = new HashSet<String>(searchNode.getStatementIds());
+                    // String statementId = searchNode.getStatementIds().get(0);
+                    Set<Node> associatedNodes = new HashSet<Node>();
+                    for (Map.Entry<String, Node> entry: transformedNetwork.getNodes().entrySet()) {
+                        // check if first statementId is in all the nodes
+                        Node n = entry.getValue();
+                        for (String statementId: n.getStatementIds()) {
+                            if (searchStatements.contains(statementId)) {
+                                searchNodes.put(entry.getKey(), n);
+                                associatedNodes.add(n);
+                            }
+                        }
+                    }
```

Comment:
Hm, so here is what I would do to avoid having to do this for every node in every network:

1. modify the class TransformedNetwork to also have a map that holds the nodes for each statementid in the network (this might be useful for other things anyway).
2. remove the setNodes method for TransformedNetwork and have instead a addNode method that adds a node to a list and also adds the node to the list of the corresponding statement id
3. add a "clear" method to remove all nodes
4. make the getNodes method return an immutable list 
5. adjust the places where nodes are added to the network to use the new methods
6. use the map<statementId, nodeslist> here instead of these two for loops

this will make your code shorter and easier to read